frontend/README.md

```diff
@@ -1,114 +1,116 @@
+
+The frontend has the following views:
+
+- **Homepage view**: this is the first page you would see when visiting the frontend. You can start the process of creating a link-bundles from here.
+
+  ![Homepage picture](docs/Images/Homepage.png)
+
+- **Edit view**: this is the page where you would create your lists.
+  
+  ![Edit picture](docs/Images/Edit_page.png)
+
+- **User view**: this page would list all list-bundles created for a signed-in user.
+  
+  ![Manage lists picture](docs/Images/Manage_lists_page.png)
+
+- **List view**: this page is used to view the list-bundle for a given vanity URL.
+
+  ![View list picture](docs/Images/View_page.png)
+
+The frontend for this project was built with the following libraries and frameworks:
+
+- [TypeScript](https://www.typescriptlang.org/)
+- [Vue.js](https://github.com/vuejs/vue) / [Vue CLI](https://github.com/vuejs/vue-cli)
+- [Vuelidate](https://github.com/vuelidate/vuelidate)
+- [Axios](https://github.com/axios/axios)
+
+Other useful tools
+
+- [Visual Studio Code](https://code.visualstudio.com/?WT.mc_id=theurlist-github-buhollan)
+- [Vetur](https://marketplace.visualstudio.com/items?itemName=octref.vetur&WT.mc_id=theurlist-github-buhollan)
+- [VS Code Debugger for Chrome](https://marketplace.visualstudio.com/items?itemName=msjsdiag.debugger-for-chrome&WT.mc_id=theurlist-github-buhollan)
+- [Vue VS Code Extension Pack](https://marketplace.visualstudio.com/items?itemName=sdras.vue-vscode-extensionpack&WT.mc_id=theurlist-github-buhollan)
+- [Vue browser devtools](https://github.com/vuejs/vue-devtools)
+
+## <a name="feconfigs" ></a> Frontend configurations
+
+There are two configurations needed for the frontend to run, those are passed as [VUE environment variables](https://cli.vuejs.org/guide/mode-and-env.html)
+
+- VUE_APP_BACKEND: this has the URL pointing to the backend endpoint.
+- VUE_APP_FRONTEND: this has the URL point to the frontend endpoint.
+- VUE_APP_OIDC_AUTHORITY: the configured Azure B2C Open ID endpoint.
+- VUE_APP_OIDC_CLIENT_ID: the configured Azure B2C Client Application ID.
+- VUE_APP_OIDC_SCOPE: the Azure B2C Client Scope.
+
+## Build and run the frontend locally
+
+### Modify Environment Variable
+
+- Follow the guide on how to run the Backend locally. You can find the README [here](../api/README.md)
+- Once the Backend has started, you will want to get the port the Backend uses _(local port may change depending upon the IDE being used)_.
+- Create a local configuration file called `env.development` and add the needed configs as described [above](feconfigs), something like the following:
+
+```bash
+VUE_APP_BACKEND=http://localhost:[backend_port]
+VUE_APP_FRONTEND=http://localhost:[frontend_port]
+VUE_APP_OIDC_AUTHORITY=[Azure B2C Open ID config endpoint]
+VUE_APP_OIDC_CLIENT_ID=[Azure B2C Client Application ID]
+VUE_APP_OIDC_SCOPE=[Azure B2C Client Scope]
+```
+
+### Install Vue CLI globally

```

Comment:
(Minor) Just to be more concise, I prefer if we just have a sequential of commands with maybe end of line comments rather than a subsection for each of them. Seems a bit overkill to me. Also, people should just search if they don't know what these commands do.

Reply:
This is the existing contents from the original authors, I didnt change them. I merged the first two to setup NPM. But the rest are used for different purposes so I kept those separate.